src/niivue.js

```diff
@@ -3254,11 +3258,153 @@ Niivue.prototype.loadDocument = function (document) {
     log.debug(meshToAdd);
     this.addMesh(meshToAdd);
   }
+
+  // handle older documents that don't have options/scene fields defined
+  this.scene = { ...this.scene, ...document.scene.sceneData };
+  this.opts = { ...this.opts, ...document.opts };
+
   this.updateGLVolume();
   this.onDocumentLoaded(document);
   return this;
 };
 
+/**
+ * generates JavaScript to load the current scene as a document
+ * @param {string} canvasId id of canvas NiiVue will be attached to
+ * @example
+ * const javascript = this.generateLoadDocumentJavaScript("gl1");
+ * const html = `<html><body><canvas id="gl1"></canvas><script type="module" async>        
+        ${javascript}</script></body></html>`;
+ */
+Niivue.prototype.generateLoadDocumentJavaScript = function (canvasId) {
+  this.document.opts = this.opts;
+  let json = this.document.json();
+  json.sceneData = { ...this.scene };
+  delete json.sceneData["sceneData"];
+  delete json.sceneData["onZoom3DChange"];
+  delete json.sceneData["onAzimuthElevationChange"];
+
+  let docString = JSON.stringify(json);
+  const buf = fflate.strToU8(docString, { level: 6, mem: 4 });
+  const compressed = fflate.compressSync(buf);
+
+  // https://stackoverflow.com/questions/68849233/convert-a-string-to-base64-in-javascript-btoa-and-atob-are-deprecated
+  const base64 = NVUtilities.uint8tob64(compressed);
+  const javascript = `
+  import * as niivue from "https://niivue.github.io/niivue/features/niivue.es.js";
```

Comment:
@cdrake , Ideally it would be nice to avoid depending on CDN imports for the saved HTML. The saved HTML files will depend on an internet connection to display the contents right?

It's possible we could add a new `define` key [here](https://github.com/niivue/niivue/blob/3c3fd2bd836deb028c6737e065e86d8bbf9bbac6/vite.config.js#L7) that would inject the compiled niivue bundle (either the .es or .umd file). This will bloat the final output (double in size) but will allow niivue to inject its own code without a CDN. 

For `fflate` could we just have an alias fflate call in niivue (`Niivue.prototype.<insert_fflate_function>`) that utilises the fflate we are already importing as part of niivue?

Reply:
Sounds good to me. I think it would be nice for users because the documents would stay functional even with major changes to NiiVue.